### DISCUSSION TITLE
GAM in R

### DISCUSSION THREAD

**Discussion by sanaashrafali** [now-i-know-my-abcd/docs#32] [Stats & Code Help]
Hi all, I'm new to GAM in R and I keep receiving this error: "'names' attribute [1] must be the same length as the vector [0]" from the following code: gam(y ~ s(x), data=data, method="REML"). I know this error isn't unique to gam() and that there is some sort of mismatch in my variables but i'm having difficulty figuring out what exactly the "names" and vector attributes are within the gam context. Any help is much appreciated!

**Comment by mrpeverill** [comment]
@sanaashrafali I've seen this error message come up a few times. From what I can tell, mgcv tends to kick it when it is having some problem with the data (for example, if x is a character vector rather than a factor). Certainly, I don't think there is anything obvious about the code you posted that would produce an error. I wish the mgcv developers had programmed in a more descriptive message.

Possibly I could help more if I new more about the data frame. Are you able to post 'data', or a subset of it? Failing that, could you post output from str(data[,c("x","y"])?
